src/main/java/bookface/model/book/BookList.java

```diff
@@ -141,6 +141,8 @@ private boolean booksAreUnique(List<Book> books) {
      */
     public void loan(Person person, Book book) {
         CollectionUtil.requireAllNonNull(person, book);
-        book.loanTo(person);
+        book.loanTo(person, book);
+        int index = internalList.indexOf(book);
```

Comment:
I think we can use JavaFX's https://docs.oracle.com/javase/8/javafx/api/javafx/collections/FXCollections.html#observableArrayList-javafx.util.Callback- here to track updates to elements of the Observable List, instead of `indexOf` which runs in linear time

Reply:
Yup that makes sense